src/blockcrawler/evm/rpc.py

```diff
@@ -195,6 +195,118 @@ async def get_block(self, block_num: HexInt, full_transactions: bool = False) ->
         )
         return block
 
+    async def get_block_by_timestamp(self, timestamp: HexInt) -> EvmBlock:
+        """Get block number by timestamp.
+
+        :param timestamp: The timestamp of the block you wish to get.
+        :returns: The block which timestamp is equal to given timestamp
+            or the nearest block if there's no block with the exact timestamp.
```

Comment:
please specify if the nearest block is before or after the timestamp.

Reply:
It returns the nearest block, no matter if it's before or after the exact timestamp.